frontend/src/pages/SubProjects/SubprojectDialog.js

```diff
@@ -58,6 +58,9 @@ const SubprojectDialog = props => {
     subProjects
   } = props;
   const changes = compareObjects(subProjects, subprojectToAdd);
+  if (changes.deletedProjectedBudgets && changes.deletedProjectedBudgets.length === 0) {
```

Comment:
the sideEffect happens at SubProjects/reducer.js -Line 60
This delete here is because deletedProjectedBudgets is default an empty array. Suggestion: use undefined as default value.

Reply:
This is necessary since `subProjects` does not have a defaultvalue for `deletedProjectedBudgets `, but `subprojectToAdd` has.
The reason it is deleted here: _isEmpty(changes) return false if there is an empty array inside

Reply:
Resolved in Commit "Review Changes 3"